Sources/SwiftSyntax/Syntax.swift

```diff
@@ -13,7 +13,7 @@
 /// A Syntax node represents a tree of nodes with tokens at the leaves.
 /// Each node has accessors for its known children, and allows efficient
 /// iteration over the children through its `children` property.
-public struct Syntax: SyntaxProtocol {
+public struct Syntax: SyntaxProtocol, SyntaxHashable {
```

Comment:
It's not clear to me, why is `SyntaxHashable` a separate protocol than `SyntaxProtocol`? Is there something that needs to conform to `SyntaxProtocol` but not `SyntaxHashable` or vice-versa?

Reply:
Hashable has a `Self` requirement and if we make `SyntaxProtocol` conform to `Hashable`, we cannot use it as a return type anymore.